### PR TITLE
Add "import" flow in the new site button

### PR DESCRIPTION
**Pull request by wojtekn**
#### Proposed Changes

* We are adding a dropdown menu link to the new site button on the Sites Management Page. The link points to the new site import flow.

Screen Shot 2022-09-15 at 10 22 43

#### Testing Instructions

1. Open the `/sites` page
2. Click the arrow in the "Add new site" button
3. Click "Import an existing site"
4. Choose a new free domain
5. On the "Choose a plan" page, select a free plan

Expected result:
* the "What are your goals?" steps is displayed
* "Import my existing website content" is checked

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/67534

### CODE REVIEW

**Comment by wojtekn** [comment]
We have the following requirement:

>  It's possible using tracks data to see that the origin of an "import" was the SMP

Currently, `calypso_signup_goals_select` is logged when we submit the Goals step and it includes the `ref` event property set to `smp-import`. Do we need more tracking, e.g., on the button click level, or is this one enough?

I have also some concerns regarding the icon. I used `arrow-down` which was proposed in early design. Should we try another one that would fit better?

I tried those:

`popout`

popout

`create`

create

`cloud-download`

cloud-download

Those are not perfect. I think that `external` icon could be a good fit. It doesn't exist in the GridIcons package yet, but it would look like `internal` icon with an arrow placed in the opposite direction:

external

What do you think?

File: client/sites-dashboard/components/sites-dashboard.tsx [hunk 1]
```diff
@@ -199,6 +199,9 @@ export function SitesDashboard( {
 							<JetpackLogo className="gridicon" size={ 18 } />
 							<span>{ __( 'Add Jetpack to a self-hosted site' ) }</span>
 						</PopoverMenuItem>
+						<PopoverMenuItem href="/start?source=sites-dashboard&ref=smp-import" icon="arrow-down">
```
Comment (sejas) [inline client/sites-dashboard/components/sites-dashboard.tsx]: @wojtekn, what do you think if we use the `TRACK_SOURCE_NAME` variable?

```diff
- <PopoverMenuItem href="/start?source=sites-dashboard&ref=smp-import" icon="arrow-down">
+ <PopoverMenuItem href={ `/start?source=${ TRACK_SOURCE_NAME }&ref=smp-import` } icon="arrow-down">
```

**Review by sejas — COMMENTED** [review]
Looking good, and testing good.
Well done adding a new ref to identify this flow.

I suggested a small change.

File: client/sites-dashboard/components/sites-dashboard.tsx (hunk 1 above)
Comment (vykes-mac) [reply client/sites-dashboard/components/sites-dashboard.tsx]: can probably use addQueryArgs also to clean-up url: `href={ addQueryArgs( { search: this.props.searchTerm }, '/sites' ) }`

**Comment by sejas** [comment]
@wojtekn there is a wording suggestion here: pdKhl6-Jm-p2

**Comment by vykes-mac** [comment]
@wojtekn 

> Currently, calypso_signup_goals_select is logged when we submit the Goals step and it includes the ref event property set to smp-import. Do we need more tracking, e.g., on the button click level, or is this one enough?

I think this track should be enough, I don't see where tracking individual button clicks makes a difference unless it makes it easier to filter and extract the track information.

> What do you think?

Personally I'm okay with the `arrow-down`, the wording suggestion posted by @sejas might result in an icon change if necessary, probably should reach out to @SaxonF on the icon issue.

File: client/sites-dashboard/components/sites-dashboard.tsx (hunk 1 above)
Comment (wojtekn) [reply client/sites-dashboard/components/sites-dashboard.tsx]: @sejas @vykes-mac thanks, I've changed how we generate all three `href` attributes and used the `addQueryArgs` function. I've re-used the existing constant as well.

File: client/sites-dashboard/components/sites-dashboard.tsx [hunk 2]
```diff
@@ -191,14 +191,29 @@ export function SitesDashboard( {
...
 						</PopoverMenuItem>
+						<PopoverMenuItem
+							href={ addQueryArgs( '/start', {
+								source: TRACK_SOURCE_NAME,
+								ref: 'smp-import',
+							} ) }
+							icon="arrow-down"
+						>
+							<span>{ __( 'Import an existing site' ) }</span>
+						</PopoverMenuItem>
```
Comment (sejas) [inline client/sites-dashboard/components/sites-dashboard.tsx:216]: Great refactor 💪

**Review by sejas — APPROVED** [review]
Looks great!

Much more clear !! ❤️ 

I would deploy with the current copy and update it later.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7531584

Thank you @wojtekn for including a screenshot in the description! This is really helpful for our translators.

**Comment by vindl** [comment]
> Currently, calypso_signup_goals_select is logged when we submit the Goals step and it includes the ref event property set to smp-import. Do we need more tracking, e.g., on the button click level, or is this one enough?

This is not visible in the Tracks graph currently without downloading the CSV and inspecting it manually. I'd add separate click events for both dropdown buttons.

**Comment by wojtekn** [comment]
@vindl I'm adding this in https://github.com/Automattic/wp-calypso/pull/67973

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7531584).

**Pull request merged**